packages/client/src/Components/Mapping/components/FormChildPerception/index.jsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react'
+import _ from 'lodash'
+
+const Perception = ({ store, onStoreUpdate, callbackFunction }) => {
+  const [currentForce] = useState(store.currentForce)
+  const [currentMarkerName] = useState(store.currentMarkerName)
+  const [forcePerception, setForcePerception] = useState(store.perception.force)
+  const [typePerception, setTypePerception] = useState(store.perception.type)
+  const [allForces] = useState(store.allForces)
+  const [allPlatforms] = useState(store.allPlatforms)
+
+  // A copy of the store to capture the updates
+  const newStore = store
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    callbackFunction(newStore)
+  }
+
+  const handleRadioChange = ({ target }) => {
+    setForcePerception(target.value)
```

Comment:
I'm probably mistaken, but aren't we storing the data twice. Do we need to `setForcePerception` _and_ `onSToreUpdate`?

Reply:
State and store are two different things, both are needed :)

Reply:
I've had a play.  

1. Without `setForcePerception`, the UI doesn't update
2. Without `newStore.perception.force = target.value`, the counter doesn't update
3. But, we can remove `onStoreUpdate` without any impact.